harvester.go

```diff
@@ -98,6 +99,25 @@ func (b *Builder) WithConsulMonitor(addr, dc, token string, timeout time.Duratio
 	return b
 }
 
+// WithConsulMonitorFromConfig enables support for monitoring key/prefixes on consul.
+//
+// It differs from WithConsulMonitor as it automatically parses the config and monitors every field tagged with consul.
+func (b *Builder) WithConsulMonitorFromConfig(addr, dc, token string, timeout time.Duration) *Builder {
```

Comment:
Since we have WithConsul, why we do not reuse this one. This would make the API clear and one way..

Reply:
Do you suggest to replace `WithConsulMonitor` entirely and only keep one method?

I didn't want to introduce breaking changes but I'm fine with it.

Reply:
Fixed - please double check